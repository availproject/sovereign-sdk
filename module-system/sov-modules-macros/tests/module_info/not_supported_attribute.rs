use sov_modules_api::{Context, ModuleInfo};
use sov_state::StateMap;

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    #[address]
    address: C::Address,

    #[other]
    test_state1: StateMap<u32, String>,

    #[state]
    test_state2: StateMap<C::PublicKey, String>,
}

fn main() {}
