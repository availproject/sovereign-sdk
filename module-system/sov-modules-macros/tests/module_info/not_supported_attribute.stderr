error: Only `#[module]`, `#[state]` or `#[address]` attributes are supported.
  --> tests/module_info/not_supported_attribute.rs:10:5
   |
10 |     test_state1: StateMap<u32, String>,
   |     ^^^^^^^^^^^

error: cannot find attribute `other` in this scope
 --> tests/module_info/not_supported_attribute.rs:9:7
  |
9 |     #[other]
  |       ^^^^^
